package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -499,17 +537,50 @@ def guess_acceptors(self, select='all', max_charge=-0.5):
         in hydrogen bonding. This :class:`str` may then be modified before being used to set the attribute
         :attr:`acceptors_sel`.
 
+        .. versionchanged:: 2.4.0
+            Added ability to use atom types
+
         """
 
         ag = self.u.select_atoms(select)
         acceptors_ag = ag[ag.charges < max_charge]
-        acceptors_list = np.unique(
-            [
-                '(resname {} and name {})'.format(r, p) for r, p in zip(acceptors_ag.resnames, acceptors_ag.names)
-            ]
-        )
 
-        return " or ".join(acceptors_list)
+        return self.__group_categories(acceptors_ag)
+
+    @staticmethod
+    def __group_categories(group):
```

Comment:
I don't see a need to make it double-underscore. Just make it single underscore, that's enough to mark it as "private" and not part of the public API. This would also make it easier to test this fragile method by itself.